Refactor/profanity checks

jmwill86 opened the pull request:
Fix/refactor based on user feedback on Discord:

> After upgrading shuttle to 0.5.0, I get an error when trying to do anything (deploy, status, local run, etc.). It tells me, that the name I set is an invalid project name. It does only contain lowercase letters, however, it contains the letters "ass" inside the Word ('dachterasse') so it might trigger the profanity check by mistake. Could this be fixed? 

I've removed the Censor::Zealous words as they didn't seem overly terrible in my opinion? I've removed some of the shorter words that might easily be within other words or phrases.

I'm not sure what the below line is/was for, it could be that it's a performance thing and my alternate way isn't ideal for some reason. I'll leave that to someone who knows more than I do on that front! Happy to make the changes though.

```
static INSTANCE: OnceCell<HashSet<String>> = OnceCell::new();
INSTANCE.get_or_init(|| HashSet::from(["Shuttle.rs".to_string()]));
```

oddgrd commented:
Nice! I agree that the Zealous word list is a bit much, causing too many false positives. 

Regarding the `OnceCell`, it's used so that the reserved word `HashSet` only gets instantiated the first time the `is_profanity_free_and_not_reserved` function is called, not every time. Here is a [relevant thread](https://users.rust-lang.org/t/lazy-static-const-fn/35870) thread from the Rust users forum (`lazy_static` offers the same functionality we used `once_cell` for here, but for a few [reasons](https://docs.rs/once_cell/1.13.1/once_cell/#faq) `once_cell` is recommended now). :slightly_smiling_face:

jmwill86 commented:
I've added Rustrict to replace Censor. The reserved 'Shuttle.rs' is covered by other lowercase and punctuation restrictions so I've removed that but left the test in incase it's an issue in future. I've also removed some unused Crates from the common Cargo.toml, they don't seem to be being used anywhere.

Thanks @oddgrd , it's a big help having some references included there to learn more from 😄

chesedo commented on common/src/project.rs:
```diff
@@ -139,6 +133,9 @@ pub mod tests {
             "myassets",
+            "scunthorpe",
+            "dachterrasse",
+            "dachterasse"
```
Love this :heart_eyes:

chesedo commented on common/src/project.rs:
```diff
@@ -51,20 +50,15 @@ impl ProjectName {
-                - Censor::Custom(allowed_words);
-            !censor.check(hostname)
+        fn is_profanity_free(hostname: &str) -> bool {
+            let (_censored, analysis) = Censor::from_str(hostname).censor_and_analyze();
+            !analysis.is(Type::MODERATE_OR_HIGHER)
```
Is it possible to get the list of reserved words back somehow?

chesedo replied:
... this is more for when we want to others in the future...

chesedo commented on common/Cargo.toml:
```diff
@@ -7,13 +7,9 @@ description = "Common library for the shuttle platform (https://www.shuttle.rs/)
-once_cell = "1.13.0"
 rocket = "0.5.0-rc.2"
 serde = "1.0.137"
-serde_json = "1.0.81"
```
Ahhh, some useless code. Nice!

chesedo reviewed (commented):
Just one question about reserved words.

`cargo sort --workspace` should fix the CI failure :smile:

jmwill86 commented:
Reserved words added back in. I've left "shuttle.rs" in for now as an example. Would it be better to just leave it blank? With us not having any reserved words currently there's no examples to add in for testing purposes either.

chesedo reviewed (approved):
Nice fixup @jmwill86! I think keeping shuttle.rs as a reserved word is perfect - we don't want it to be a project name and shouldn't rely on some other check which implicitly covers it
